docs/how_to_use_csmctl.md

```diff
@@ -0,0 +1,77 @@
+# Using csmctl
+
+
+## What does csmctl do? 
+
+As a user, you can create clusters based on Cluster Stacks with the help of the Cluster Stack Operator. The operator needs certain files, e.g. to apply the required Helm charts, and to get the necessary information about the versions in the cluster stack.
+
+In order to not generate these files manually, this CLI tool takes a certain pre-defined directory structure, in which users can configure all necessary Helm charts and build scripts for node images, and generates the assets that the Cluster Stack Operator can process.
+
+Therefore, this tool can be used to configure Cluster Stacks and to test them with the Cluster Stack Operator. It can also be used to release stable releases of Cluster Stacks that can be published for a broader community.
+
+## Different modes of csmctl 
+
+The csmctl has multiple modes that can be used for different use cases.
+
+### Hash mode
+
+This mode is the most used one, as it allows quick iterations and testing of a cluster stack. It takes the hash of the content of the cluster stack and generates a semver version on this. You can combine it with the `custom` channel of Cluster Stack Operator and test your Cluster Stacks easily!
+
+### Stable mode
+
+This mode checks for existing releases of cluster stacks and versions your cluster stack accordingly. If you have an existing release of "v1", then it would use "v2" for the new one. It also checks whether the node images and cluster addons have changed or not and will only update the versions if something actually changed.
+
+### Beta mode
+
+Similar to stable mode, but for a beta release channel. It versions according to "v0-beta.1", etc.
+
+### Custom mode
+
+The custom mode can be used to define your own version. You can input any semver version and your cluster stack will be versioned accordingly.
+
+
+## Installing csmctl
+You can click on the respective release of csmctl on GitHub and download the binary.
+
+Assuming, you have downloaded the `csmctl_0.0.2_linux_amd64` binary in your Downloads directory, you will need the following commands to rename the binary and to give it executable permissions.
+
+```bash
+$ sudo chmod u+x ~/Downloads/csmctl_0.0.2_linux_amd64
+$ sudo mv ~/Downloads/csmctl_0.0.2_linux_amd64 /usr/local/bin/csmctl # or use any bin directory from your PATH
+```
+
+Then you can check whether everything worked by printing the version of csmctl.
+
+```bash
+$ csmctl version
+csmctl version: 0.0.2
+commit: f252304eb013014b35f8a91abf1f61aff2062601
```

Comment:
Hi @kranurag7 and @janiskemper, I can't find the csmctl release 0.0.2. 
I tried building it with `go build`, but the `version` command is not available there:
```
./csmctl version
Error: unknown command "version" for "csmctl"
Run 'csmctl --help' for usage.
```
Did I do something wrong in building or is that an old one?
I'm in the branch `syself/add-csmctl-docs`,  (`commit 0e908417b2116d06b5d6f997b8d5f9323639e3b8`)

The `create` command seems to work fine

Reply:
yes, we missed that: https://github.com/SovereignCloudStack/csmctl/issues/67

Reply:
this is resolved by the issue!